tests/tracer/test_logging.py

```diff
@@ -0,0 +1,356 @@
+
+
+def test_debug_logs_go_to_stderr(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
+    """
+    When setting up the default logger in debug mode without a log path,
+        it automatically logs to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"debug logs" in err
+    assert out == b""
+
+    code = """
+import logging
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"debug logs" in err
+    assert out == b""
+
+
+def test_debug_logs_can_go_to_file(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess, tmpdir):
+    """
+    When setting up the default ddtrace logger in debug mode,
+        it automatically logs to a file at the expected backup count.
+    """
+    env = os.environ.copy()
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_DEBUG"] = "true"
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "10"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 10
+assert ddtrace_logger.handlers[0].backupCount == 1
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+
+    testfiles = os.listdir(tmpdir)
+    files_created = 0
+    for file in testfiles:
+        if "testlog.log" in file:
+            files_created = files_created + 1
+
+    assert files_created == 2
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert re.search(LOG_PATTERN, content) is not None
+
+    code = """
+import logging
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 10
+assert ddtrace_logger.handlers[0].backupCount == 1
+"""
```

Comment:
```suggestion
```

Reply:
I think this change was to try to collapse the tests down, but the two sets of code snippets differ because of them does `imports ddtrace` to show that the import behavior doesn't change anything.